realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -331,6 +338,80 @@ private void emitAccessors(JavaWriter writer) throws IOException {
         }
     }
 
+    private void emitRealmObjectProxyImplementation(JavaWriter writer) throws IOException {
```

Comment:
How about just adding `getProxyState()` instead of delegating all the methods? It should be just as efficient and we would save a lot of generated methods on the proxy classes?